### PR TITLE
Pim 1038

### PR DESCRIPTION
**Pull request by pavlovdog**
- Added Trust account support

### CODE REVIEW

File: packages/permissionless-test/mock-aa-infra/alto/src/index.ts [hunk 1]
```diff
@@ -366,35 +441,13 @@ const main = async () => {
         address: kernelFactoryOwner
     })
 
-    // ==== SETUP ALCHEMY LIGHT ACCOUNT CONTRACTS ==== //
-    const alchemyLightClientOwner = "0xDdF32240B4ca3184De7EC8f0D5Aba27dEc8B7A5C"
```
Comment (plusminushalf) [inline packages/permissionless-test/mock-aa-infra/alto/src/index.ts:370]: Why have we removed this?

File: packages/permissionless/accounts/index.ts [hunk 2]
```diff
@@ -85,5 +93,6 @@ export {
     type PrivateKeyToSafeSmartAccountParameters,
     type SignerToEcdsaKernelSmartAccountParameters,
     type SignerToBiconomySmartAccountParameters,
-    type PrivateKeyToBiconomySmartAccountParameters
+    type PrivateKeyToBiconomySmartAccountParameters,
+    type SignerToTrustSmartAccountParameters
```
Comment (plusminushalf) [inline packages/permissionless/accounts/index.ts:97]: we should add `PrivateKeyToTrustSmartAccount` as well.

File: packages/permissionless/accounts/trust/signerToTrustSmartAccount.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    entryPoint extends EntryPoint,
+    transport extends Transport = Transport,
+    chain extends Chain | undefined = Chain | undefined
+> = SmartAccount<entryPoint, "TrustSmartAccount", transport, chain>
+
+export type SignerToTrustSmartAccountParameters<
+    entryPoint extends EntryPoint,
+    TSource extends string = string,
+    TAddress extends Address = Address
+> = Prettify<{
```
Comment (plusminushalf) [inline packages/permissionless/accounts/trust/signerToTrustSmartAccount.ts:53]: We don't need prettify here, it is mostly to simplify complex inferred objects.

File: packages/permissionless/accounts/trust/signerToTrustSmartAccount.ts [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        address ??
+            getAccountAddress(client, {
+                factoryAddress,
+                secp256k1VerificationFacetAddress,
+                entryPoint: entryPointAddress,
+                bytes: viemSigner.publicKey,
+                owner: viemSigner.address,
+                index
+            }),
+        getChainId(client)
```
Comment (plusminushalf) [inline packages/permissionless/accounts/trust/signerToTrustSmartAccount.ts:130]: Can we check if client has the chain so

`client.chain?.id ?? getChainId(client)`

File: packages/permissionless/accounts/trust/utils/getFactoryData.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    bytes,
+    index,
+    secp256k1VerificationFacetAddress
+}: {
+    account: Account
+    bytes: `0x${string}`
+    index: bigint
+    secp256k1VerificationFacetAddress: Address
+}) => {
+    if (!account.address) throw new Error("Owner account not found")
```
[the remark this answers is not in the document]
Comment (pavlovdog) [reply packages/permissionless/accounts/trust/utils/getFactoryData.ts]: Done

File: packages/permissionless/accounts/trust/signerToTrustSmartAccount.ts (hunk 4 above)
Comment (pavlovdog) [reply packages/permissionless/accounts/trust/signerToTrustSmartAccount.ts]: Done

**Review by plusminushalf — APPROVED** [review]

**Pull request merged**